disttask/importinto/encode_and_sort_operator.go

```diff
@@ -125,10 +169,27 @@ func (w *chunkWorker) HandleTask(task *importStepMinimalTask, _ func(workerpool.
 	// we don't use the input send function, it makes workflow more complex
 	// we send result to errCh and handle it here.
 	executor := newImportMinimalTaskExecutor(task)
-	if err := executor.Run(w.ctx); err != nil {
+	if err := executor.Run(w.ctx, w.dataWriter, w.indexWriter); err != nil {
 		w.op.onError(err)
 	}
 }
 
-func (*chunkWorker) Close() {
+func (w *chunkWorker) Close() {
+	closeCtx := w.ctx
+	if closeCtx.Err() != nil {
+		// in case of context canceled, we need to create a new context to close writers.
+		closeCtx = context.Background()
```

Comment:
maybe some timeout to avoid blocking main logic. resource leak is not that harmful